File: Question 5/is_valid_name.rb
def is_valid_name(str)
    parts = str.split(" ")
    if parts.length < 2
      return false
    end
  
    parts.each do |part|
      if !is_capitalized(part)
        return false
      end
    end
  
    return true
  end
  
  def is_capitalized(word)
    if word[0] == word[0].upcase && word[1..-1] == word[1..-1].downcase
      return true
    else
      return false
    end
  end
  
  puts is_valid_name("Kush Patel")       # => true
  puts is_valid_name("Daniel")           # => false
  puts is_valid_name("Robert Downey Jr") # => true
  puts is_valid_name("ROBERT DOWNEY JR") # => false

File: Arrays/Question 5/is_valid_name.js
function is_valid_name(str){
    let arr = str.split(' ');
  
  for (let i =0; i< arr.length; i++){
    if((arr.length >= 2) && (firstword(arr[i]) === true)){
      return true;
    }
   }
  return false;
  }
  

  //create a function to check for each words and test for uppercase
  function firstword(word){
    for(let i =1; i<word.length; i++){
      if(!(word[0] === word[0].toUpperCase() === true &&
      word[i] === word[i].toUpperCase() === false)){
        return false;
      }
    }
    return true;
  }

console.log(is_valid_name("Kush Patel")) // => true
console.log(is_valid_name("Daniel"))  // => false
console.log(is_valid_name("Robert Downey Jr"))// => true
console.log(is_valid_name("ROBERT DOWNEY JR")) // => false
